lib/linked_list.rb

```diff
@@ -18,71 +18,163 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(value)
-      raise NotImplementedError
+
+      new_node = Node.new(value, nil)
+      new_node.next = @head
+      @head = new_node
+      return @head.data

```

Comment:
This is a little simpler

```suggestion
      @head = Node.new(value, @head)
```